tests/testsuite/path.rs

```diff
@@ -589,6 +589,354 @@ Caused by:
+        .build();
+
+    let p = project()
+        .file(
+            ".cargo/config.toml",
+            &format!(
+                "[path-bases]\nworkspace = '{}/subdir'",
+                bar.root().parent().unwrap().display()
+            ),
+        )
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "parent"
+                version = "0.1.0"
+                authors = []
+
+                [workspace]
+                members = ["child"]
+
+                [workspace.dependencies.dep_with_base]
+                path = '../dep_with_base'
+                base = 'workspace'
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file(
+            "child/Cargo.toml",
+            r#"
+                [package]
+                name = "child"
+                version = "0.1.0"
+                authors = []
+                workspace = ".."
+
+                [dependencies.dep_with_base]
+                workspace = true
+            "#,
+        )
+        .file("child/src/main.rs", "fn main() {}");
+    let p = p.build();
+
+    p.cargo("build -v -Zpath-bases")
+        .masquerade_as_nightly_cargo(&["path-bases"])
+        .run();
+}
+
+#[cargo_test]
+fn unknown_base() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+
+                [dependencies.bar]
+                path = 'bar'
+                base = 'test'
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("build -Zpath-bases")
+        .masquerade_as_nightly_cargo(&["path-bases"])
+        .with_status(101)
+        .with_stderr_data(
+            "\
+[ERROR] failed to parse manifest at `[..]/foo/Cargo.toml`
+
+Caused by:
+  resolving path base for dependency (bar)
+
+Caused by:
+  path base `test` is undefined. You must add an entry for `test` in the Cargo configuration [path-bases] table.
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn workspace_builtin_base_not_a_workspace() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+
+                [dependencies.bar]
+                path = 'bar'
+                base = 'workspace'
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("build -Zpath-bases")
+        .masquerade_as_nightly_cargo(&["path-bases"])
+        .with_status(101)
+        .with_stderr_data(
+            "\
+[ERROR] failed to parse manifest at `[..]/foo/Cargo.toml`
+
+Caused by:
+  resolving path base for dependency (bar)
+
+Caused by:
+  the `workspace` built-in path base cannot be used outside of a workspace.
+",
+        )
```

Comment:
Cargo does see this as a workspace.  I'm unsure if its worth preventing this or not

Reply:
Maybe I'm misunderstanding what it means to be "in a workspace" - it would certainly simplify the logic of finding a workspace root.

When I look at the docs, it mentions that a workspace is created using the `[workspace]` table, so I find it surprising that this is considered a workspace.

Reply:
Every package has a workspace, some are just implicitly created.

Reply:
Fair enough - I've removed this error

Reply:
We should still have a test for this, either way.

I've noted the direction *I think* you went in #14355 for evaluation during stabilization

Reply:
Even though I removed the error, it looks like the code that finds a workspace root already raises an error for this case, so I'm going to stick with using that error. I've removed the note from the tracking issue.

Reply:
ok

I've added it as an unresolved question